### ISSUE TITLE
See "cached" memory

### ISSUE THREAD

**Issue by stefan-reich** [oshi/oshi#424]
Hi, I love this stuff, am using it [a lot](http://tinybrain.de/1010947).

Question: On Linux in top, there is:

KiB Mem :  1999172 total,  1476732 free,   268320 used,   254120 buff/cache
KiB Swap:   121852 total,   121852 free,        0 used.  1569708 avail Mem 

Can I get the "buff/cache" value through oshi? That would be amazing as that really defines the free memory too (caches are essentially free memory).

Cheers

**Comment by dbwiddis** [comment]
Thanks for the compliment!  And thanks for your interest in how OSHI reports memory... my original contribution to this project was because I was frustrated with how the Java OperatingSystemMXBean handled free memory... so of course accuracy here is near and dear to my heart.

Because OSHI is cross-platform, we want to keep statistics cross-platform as well, which means we conform the Linux statistics to match the other OS's.

The free memory is a bit more complicated than adding and subtracting the numbers on `top`.  For a full explanation, see [here](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773).  OSHI uses the MemAvailable calculation in the Linux Kernel for the most accurate report, if it's available.  If not, it still tries to come up with a reasonable estimate that includes "reclaimable" memory.

In summary, what OSHI says is "free" is more than the "free" you see on Top. It includes the reclaimable portions of the buff/cache section as well.

**Comment by stefan-reich** [comment]
Oh... then it's solved, right? lol. Cool
